src/bot/commands/misc/convert.js

```diff
@@ -27,12 +27,50 @@ async function fetchLatestRates () {
  * @param {number} baseValue
  * @param {string} baseType Base unit
  * @param {string} targetType Target unit
- * @returns {strung} Success message
+ * @returns {string} Success message
  * @throws {Error} Generic error if conversion fails
  */
 function convertUnits (baseValue, baseType, targetType) {
 	const conversion = convert(baseValue).from(baseType).to(targetType);
-	return `${baseValue}${baseType} is equal to ${conversion.toFixed(2)}${targetType}`;
+	const fractionDigits = getFractionDigits(conversion);
+	return `${baseValue.toString()}${baseType} is equal to ${conversion.toFixed(fractionDigits)}${targetType}`;
+}
+
+/**
+ * Gets the count of zeroes after the decimal point but before a non-zero number
+ * @param {number} value
+ * @returns {number} Number of zeroes
+ */
+function getCountOfZeroesAfterDecimalPoint (value) {
+	let fractionalPart = value % 1;
+	let numberOfZeroes = -1;
+
+	while (fractionalPart < 1 && fractionalPart > 0) {
+		fractionalPart *= 10;
+		numberOfZeroes++;
+	}
+	numberOfZeroes = Math.max(0, numberOfZeroes);
+
+	return numberOfZeroes;
+}
+
+/**
+ * Gets the number of fraction digits needed to display our result with toFixed
+ * @param {number} value
+ * @returns {number} Number of fraction digits
+ */
+function getFractionDigits (value) {
+	const numberOfZeroes = getCountOfZeroesAfterDecimalPoint(value);
+
+	let fractionDigits = 2;
+	if (numberOfZeroes > 0) {
+		fractionDigits = numberOfZeroes + 2;
+	}
+	// Since toFixed doesn't accept values greater than 20
```

Comment:
Interesting, didn't know this. Is this documented somewhere? Mind linking it if you have it handy?

Reply:
[Here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed)

> The number of digits to appear after the decimal point; this may be a value between 0 and 20, inclusive, and implementations may optionally support a larger range of values. If this argument is omitted, it is treated as 0